File: kinotic-js/workspace/packages/management-api/src/api/model/grind/events/TaskStartedEvent.ts
import { JobRunEventType } from '@/api/model/grind/events/JobRunEventType'

/**
 * A task began executing.
 */
export class TaskStartedEvent {

    public readonly type: JobRunEventType.TASK_STARTED = JobRunEventType.TASK_STARTED

    /**
     * The task's position in the run's task tree.
     */
    public readonly taskPath: string = ''

    /**
     * The task's description.
     */
    public readonly description: string | null = null

}

File: kinotic-grind/src/main/java/org/kinotic/grind/api/model/events/TaskStartedEvent.java
package org.kinotic.grind.api.model.events;

/**
 * A task began executing.
 *
 * @param taskPath    the task's position in the run's task tree
 * @param description the task's description
 */
public record TaskStartedEvent(String taskPath, String description) implements JobRunEvent {
}
